llvm/include/llvm/DebugInfo/LogicalView/Core/LVObject.h

```diff
@@ -246,20 +246,17 @@ class LVObject {
   virtual void setName(StringRef ObjectName) {}
 
   LVElement *getParent() const {
-    assert((!Parent.Element ||
-            (Parent.Element && static_cast<LVElement *>(Parent.Element))) &&
+    assert((!Parent.Element || static_cast<LVElement *>(Parent.Element)) &&
```

Comment:
Upon another look, the original assert (before this mod) looks flawed.
`(! Parent.Element)` always evaluates to `true` except for when `Parent.Element` is an all-1 bitvector..
Could you update this and the other 2 assert conditions like below?

```suggestion
    assert(((Parent.Element == nullptr) || static_cast<LVElement *>(Parent.Element)) &&
```

Reply:
Oops, please disregard my previous comment.. `Parent.Element` is converted to bool before the `!` operator is applied, so there aren't any issues with the original assert.

Reply:
No problem!